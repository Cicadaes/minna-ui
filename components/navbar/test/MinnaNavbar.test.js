'use strict';

const MinnaNavbar = require('../src/MinnaNavbar.html');

const menuItems = [
  { url: '/page-one', name: 'Page One' },
  { url: '/page-two', name: 'Page Two' },
  { url: '/about', name: 'About Us' },
];

describe('MinnaNavbar component', () => {
  it('should render correctly', () => {
    const target = document.createElement('div');
    new MinnaNavbar({
      target,
      data: {
        items: menuItems,
        page: '/',
      },
    });
    expect(target.innerHTML).toMatchSnapshot();
  });
});
