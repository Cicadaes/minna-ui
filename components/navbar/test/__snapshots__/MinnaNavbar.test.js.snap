// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MinnaNavbar component can dynamically add menu items 1`] = `
"<header class=\\"navbar \\"><nav class=\\"dfc fww con\\" role=\\"navigation\\"><button type=\\"button\\" class=\\"navbar-button dn-l button-clear\\" aria-label=\\"menu toggle\\" aria-expanded=\\"false\\"><svg class=\\"navbar-icon\\"><use xlink:href=\\"#menu\\"></use></svg></button>

    <a href=\\"/\\" class=\\"lh1\\" title=\\"Home\\"><svg class=\\"navbar-logo\\"><use xlink:href=\\"#logo\\"></use>
      </svg></a>

    <div class=\\"navbar-links dn\\"><a href=\\"/\\" class=\\"navbar-link dn-l\\">Home</a>

      <a href=\\"/page-one\\" class=\\"navbar-link \\">Page One</a><a href=\\"/page-two\\" class=\\"navbar-link \\">Page Two</a><a href=\\"/page-two/child-one\\" class=\\"navbar-link \\">Child One - Page Two</a><a href=\\"/page-two/child-two\\" class=\\"navbar-link \\">Child Two - Page Two</a><a href=\\"/about\\" class=\\"navbar-link \\">About Us</a><a href=\\"page-new\\" class=\\"navbar-link \\">Page New</a></div></nav></header>"
`;

exports[`MinnaNavbar component renders correctly with required props set 1`] = `
"<header class=\\"navbar \\"><nav class=\\"dfc fww con\\" role=\\"navigation\\"><button type=\\"button\\" class=\\"navbar-button dn-l button-clear\\" aria-label=\\"menu toggle\\" aria-expanded=\\"false\\"><svg class=\\"navbar-icon\\"><use xlink:href=\\"#menu\\"></use></svg></button>

    <a href=\\"/\\" class=\\"lh1\\" title=\\"Home\\"><svg class=\\"navbar-logo\\"><use xlink:href=\\"#logo\\"></use>
      </svg></a>

    <div class=\\"navbar-links dn\\"><a href=\\"/\\" class=\\"navbar-link dn-l\\">Home</a>

      <a href=\\"/page-one\\" class=\\"navbar-link \\">Page One</a><a href=\\"/page-two\\" class=\\"navbar-link \\">Page Two</a><a href=\\"/page-two/child-one\\" class=\\"navbar-link \\">Child One - Page Two</a><a href=\\"/page-two/child-two\\" class=\\"navbar-link \\">Child Two - Page Two</a><a href=\\"/about\\" class=\\"navbar-link \\">About Us</a></div></nav></header>"
`;
